tests/types.lisp

```diff
@@ -17,6 +17,19 @@
   (is (typep (list :name 1 #\a 'x)
              '(tuple :name 1 #\a 'x))))
 
+(test function-type-declaration
+  (finishes
+    (let ((function-name-1 (gensym))
+	  (function-name-2 (gensym))
+	  (function-name-3 (gensym))
+	  (function-name-4 (gensym))
+	  (function-name-5 (gensym))
+	  (function-name-6 (gensym)))
+      `(-> ,function-name-1 (integer) integer)
+      `(-> (,function-name-2 ,function-name-3) (integer) integer)
+      `(-> (setf ,function-name-4) (real) real)
+      `(-> (,function-name-5 (setf ,function-name-6)) (real) real))))
```

Comment:
What is this actually testing (besides creating lists)? Did you mean to macroexpand these?

Reply:
These tests were silly I have to admit. Now they use `macroexpand-1` and `equalp` to compare. We might also use `equal` for comparison if you want that.